File: game/src/main/scala/falcun/net/api/gui/effects/ScissorEffect.scala
package falcun.net.api.gui.effects

import falcun.net.api.gui.components._
import falcun.net.api.gui.effects.Effect.Phase
import falcun.net.api.gui.region.GuiRegion
import falcun.net.api.gui.util.ScissorManager

class ScissorEffect(var custom: GuiRegion) extends Effect {
  override def draw(mX: Int, mY: Int, component: Component, phase: Effect.Phase): Unit = {
    if (phase eq Phase.BEFORE) ScissorManager.startScissor(if (custom == null) component.region
    else custom)
    else ScissorManager.finishScissor()
  }
}

File: client/projects/Forge/src/main/java/falcun/net/api/gui/effects/ScissorEffect.java
package falcun.net.api.gui.effects;
import com.mojang.realmsclient.gui.ChatFormatting;
import falcun.net.api.gui.components.*;

import java.util.function.BiConsumer;
import java.util.function.Supplier;
import java.util.function.Consumer;

import falcun.net.api.gui.region.GuiRegion;
import falcun.net.api.gui.util.ScissorManager;
import net.minecraft.client.Minecraft;
import org.lwjgl.input.Mouse;
public class ScissorEffect extends Effect {

	public GuiRegion custom;

	public ScissorEffect(GuiRegion custom) {
		this.custom = custom;
	}

	@Override
	public void draw(int mX, int mY, Component component, Phase phase) {
		if (phase == Phase.BEFORE) {
			ScissorManager.startScissor(custom == null ? component.region : custom);
		} else {
			ScissorManager.finishScissor();
		}
	}

}